### ISSUE TITLE
Working with Android 12? 

### ISSUE THREAD

**Issue by gprn** [Magisk-Modules-Alt-Repo/supl-replacer#2]
Hi,
Is there a chance this script could be used with Android 12 - LOS 19.1, as I don't see any cautions under the module page?
I installed it without errors but it seems GPS still want connected to glpals.com servers.

**Comment by D3SOX** [comment]
Yes, it should work. I currently use it on LineageOS 20 (Android 13) and have used it before on 19.1 and 18.1.

Can you send your `gps.conf`? It should be either under `/system/vendor/etc/gps.conf` or if that doesn't exist `/system/etc/gps.conf`
You can obtain it via an `adb shell`, Termux or any root file explorer

**Comment by gprn** [comment]
No gps.conf file anywhere on my phone, I only have a gps_debug.conf under /system/etc/ 
Is it the one for tweaking supl server?

**Comment by D3SOX** [comment]
Not sure why that is, but I might need to create one if it doesn't exist sometimes.

**Comment by HarriBuh** [comment]
I just checked for myself on my Pixel 7 w/ Android 13 and "gps.xml" under system/vendor/etc/gnss also says "acSuplServer="supl.google.com" ". I too have no "gps.conf" anywhere on my phone. The "gps_debug.conf"- file does only contain inactive code (everything is deactivated by "#" in front of every line).

**Comment by HarriBuh** [comment]
Anybody working on a fix?

**Comment by HarriBuh** [comment]
@D3SOX Is this repo already abandoned? Looks like it.

**Comment by D3SOX** [comment]
@HarriBuh It's not abandoned. I just haven't had the time to work on this.
But anybody can help here, if someone makes a PR I'm definitely going to review and merge it.

**Comment by UjuiUjuMandan** [comment]
I can confirm the modification in /system/etc/gps-debug.conf works on my Pixel 7a on Android 14, I set it as

```
SUPL_HOST=supl.grapheneos.org
```

After reboot, this host appears instead of `supl.google.com` in PCAPdroid logs.

And have a look at this: https://review.calyxos.org/c/CalyxOS/device_google_redbull/+/17021

What are they trying to do?

**Comment by UjuiUjuMandan** [comment]
Another option is to use [Pixel IMS](https://github.com/kyujin-cho/pixel-volte-patch) to modify `gps.supl_host` value, I don't know if gps-debug.conf or this carrier config has more priority.

**Comment by HarriBuh** [comment]
@UjuiUjuMandan How do I get the supl_host log from PCAPdroid? I frequently use this app, but never tried to log this specific way.

**Comment by UjuiUjuMandan** [comment]
> @UjuiUjuMandan How do I get the supl_host log from PCAPdroid? I frequently use this app, but never tried to log this specific way.

Disable Target apps and start capture and make a locate request like open https://browserleaks.com/geo .

**Comment by gprn** [comment]
> I can confirm the modification in /system/etc/gps-debug.conf works on my Pixel 7a on Android 14, I set it as
> 
> ```
> SUPL_HOST=supl.grapheneos.org
> ```
> 
> After reboot, this host appears instead of `supl.google.com` in PCAPdroid logs.
> 
> And have a look at this: https://review.calyxos.org/c/CalyxOS/device_google_redbull/+/17021
> 
> What are they trying to do?


Yes, thank you man, had never try to modify this file but my GPS chip appears to use graphene supl  too now, as I have seen on old MatLog app.

**Comment by UjuiUjuMandan** [comment]
> it seems GPS still want connected to glpals.com servers.

This domain seems to be an XTRA server, Qualcomm's PSDA implementation, not related to SUPL at all.

People in my link want to change it from `xtracloud.net` to `gllto.glpals.com` but failed.

**Comment by harphere** [comment]
> I can confirm the modification in /system/etc/gps-debug.conf works on my Pixel 7a on Android 14, I set it as
> 
> ```
> SUPL_HOST=supl.grapheneos.org
> ```
> 
> After reboot, this host appears instead of `supl.google.com` in PCAPdroid logs.
> 
> And have a look at this: https://review.calyxos.org/c/CalyxOS/device_google_redbull/+/17021
> 
> What are they trying to do?

Hi, do you mind clarifying what exactly you did?

In the gps_debug.conf, did you just change the SUPL_HOST from supl.google.com to supl.grapheneos.org and uncomment the whole line? (currently as you stated is is commented)
Did you modify the Magisk module in any way?

**Comment by UjuiUjuMandan** [comment]
It won't get processed if commented. You can ignore all commented content and add that line at the end.

No. gps-debug.conf can be edited directly with root, no need for a module.

**Comment by themidnightmaniac** [comment]
Submitted a [pr](https://github.com/Magisk-Modules-Alt-Repo/supl-replacer/pull/4) with the option to patch gps_debug.conf on devices that have it instead of gps.conf. Now the module should work out of the box on those devices.

**Comment by owocado** [comment]
> Yes, it should work. I currently use it on LineageOS 20 (Android 13) and have used it before on 19.1 and 18.1.
> 
> Can you send your `gps.conf`? It should be either under `/system/vendor/etc/gps.conf` or if that doesn't exist `/system/etc/gps.conf` You can obtain it via an `adb shell`, Termux or any root file explorer

it's broken on my phone running Android 14 which has /system/etc/gps_debug.conf (and not gps.conf)

```ini
# Sample file for use for on device debug override only
# Prefer frameworks/base/core/res/res/values/config.xml and
# frameworks/base/core/res/res/values-mcc*-mnc*/config.xml

################################
##### AGPS server settings #####
################################
# FOR SUPL SUPPORT, set the following
# SUPL_HOST=supl.google.com or IP
# SUPL_PORT=7275

# supl version 2.0
# SUPL_VER=0x20000

#SUPL_MODE is a bit mask set in config.xml per carrier by default.
#If it is uncommented here, this value will overwrite the value from
#config.xml.
#MSA=0X2
#MSB=0X1
#SUPL_MODE=1

# Emergency SUPL, 1=enable, 0=disable
#SUPL_ES=0

#Choose PDN for Emergency SUPL
#1 - Use emergency PDN
#0 - Use regular SUPL PDN for Emergency SUPL
#USE_EMERGENCY_PDN_FOR_EMERGENCY_SUPL=0

####################################
#  LTE Positioning Profile Settings
####################################
# 0: Enable RRLP on LTE(Default)
# 1: Enable LPP_User_Plane on LTE
# 2: Enable LPP_Control_Plane
# 3: Enable both LPP_User_Plane and LPP_Control_Plane
#LPP_PROFILE = 2

##################################################
# Select Positioning Protocol on A-GLONASS system
##################################################
# 0x1: RRC CPlane
# 0x2: RRLP UPlane
# 0x4: LLP Uplane
#A_GLONASS_POS_PROTOCOL_SELECT = 0

# Below bit mask configures how GPS functionalities
# should be locked when user turns off GPS on Settings
# Set bit 0x1 if MO GPS functionalities are to be locked
# Set bit 0x2 if NI GPS functionalities are to be locked
# default - non is locked for backward compatibility
#GPS_LOCK = 0

################################
##### PSDS download settings #####
################################
# For wear devices only.
# Enable periodic PSDS download once a day.
# true: Enable periodic PSDS download
# false: Disable periodic PSDS download
#ENABLE_PSDS_PERIODIC_DOWNLOAD=false

```

**Comment by D3SOX** [comment]
I'm gonna push an update that fixes it

**Comment by D3SOX** [comment]
Use [v2.3.0](https://github.com/Magisk-Modules-Alt-Repo/supl-replacer/releases/tag/v2.3.0)

**Comment by owocado** [comment]
Thank you, have a nice cozy day <3

**Comment by D3SOX** [comment]
You too :) Can you report if it worked?

**Comment by owocado** [comment]
So with my current root setup via KernelSU + ZygiskNext, flashing [v2.3.0](https://github.com/Magisk-Modules-Alt-Repo/supl-replacer/releases/tag/v2.3.0) doesn't seem to have any effect.
So then I flashed this on my friends phone which is rooted via magisk where it works.
I'm trying to find why this doesn't work with ZygiskNext+kSU, prolly skill issue.

**Comment by D3SOX** [comment]
Hm I tried it with APatch on Lineage 23 (A16, treble version with `/system/vendor/etc/gps.conf`) and it worked fine
Where are the files on those 2 devices?

**Comment by D3SOX** [comment]
I have a feeling that it's not using `gps_debug.conf`.
It even says at the top
```
# Sample file for use for on device debug override only
```
I'm adding copying it to `gps.conf` in another release

**Comment by D3SOX** [comment]
Does [v2.3.1](https://github.com/Magisk-Modules-Alt-Repo/supl-replacer/releases/tag/v2.3.1) solve the issue?

**Comment by owocado** [comment]
> Does [v2.3.1](https://github.com/Magisk-Modules-Alt-Repo/supl-replacer/releases/tag/v2.3.1) solve the issue?

Looks like it worked yay 🎉 
Your solution to copy gps_debug.conf as gps.conf did the trick and is genius tbh. My silly brain missed that xD


<img width="1080" height="672" alt="Image" src="https://github.com/user-attachments/assets/5b31da27-37f5-4c4b-abf1-1b5434b0054c" />
<img width="1080" height="830" alt="Image" src="https://github.com/user-attachments/assets/7f494b80-745d-4028-baad-73925bbe934c" />

**Comment by D3SOX** [comment]
If you have any kind of Firewall you could now double-check if it actually applied

**Comment by HarriBuh** [comment]
It didn't work with me. 
I installed 2.3.1 via Magisk (30.4), checked inside gps.conf (system/etc; graphene os was set) and tested it via Browser (Brave + Browserleaks.com)
PCAPdroid gave me this:
![Screenshot_20251106-142814.png](https://github.com/user-attachments/assets/cfceccf5-c39e-4630-9407-ea69355a7581)

Google Pixel 9, Android 16

**Comment by D3SOX** [comment]
Then it's not using that file. I can't debug this you'll have to figure out what config your device uses.

**Comment by owocado** [comment]
@HarriBuh um how do I repro this? can you please share repro steps?
I visited browserleaks while PCAPdroid running but I don't see a request being made to supl.google.com
Maybe because I also have adguard running too? which has about 600k DNS filter entries to block? or I'm dum >.<

**Comment by HarriBuh** [comment]
> @HarriBuh um how do I repro this? can you please share repro steps?
> I visited browserleaks while PCAPdroid running but I don't see a request being made to supl.google.com
> Maybe because I also have adguard running too? which has about 600k DNS filter entries to block? or I'm dum >.<

You will need to prepare for this:
Enable location tracking on your device AND browser.
Go to the "Location API" tab on Browserleaks.com.
It (the browser) needs to ask you for this permission, otherwise you did something wrong.

**Comment by D3SOX** [comment]
I'm on lineage for microg android 16 can't also not trigger a request to any supl server using this method

**Comment by owocado** [comment]
ummmmm silly question but what's "location tracking" mean exactly here? what is it called in device settings and browser settings? ex. Chrome
Though I simply turned on location from quick settings, visited browserleaks.com/geo while PCAPdroid running.
It showed popup to ask my location, I allowed but still I don't see any request being made to supl.google.com

<details>
<summary>Screenshots</summary>
<img width="1080" height="826" alt="Image" src="https://github.com/user-attachments/assets/2a0240e4-3ee1-4327-8efd-594624b3e87c" />
<img width="1080" height="2075" alt="Image" src="https://github.com/user-attachments/assets/45211299-dae7-4f57-bba4-f35082ae7ff2" />
</details>

someone please tell me if this is sign that module is working or not LOL

**Comment by D3SOX** [comment]
A bit off topic: Why are you even using mainline Chrome if you're using this module. Try using [Cromite](https://github.com/uazo/cromite)

**Comment by owocado** [comment]
> A bit off topic: Why are you even using mainline Chrome if you're using this module. Try using [Cromite](https://github.com/uazo/cromite)

\>.< Sorry, I was just being lazy while doing repro. I daily drive firefox nightly but I wasn't sure if it will be a good candidate for repro i.e. different than Chrome/Chromium
Though Cromite gives the same result after trying, not seeing any request to supl.google.com

Even opened Google Maps app, allowed it to access my location, checked back in PCAPdroid but still the same result 🤔 
Does that mean this module is working for me?

**Comment by D3SOX** [comment]
I tried the same but I wonder why I'm not seeing one to supl.grapheneos.org either. I also tried it using Pi-Hole

**Comment by D3SOX** [comment]
Are we all using microG? @HarriBuh Please explain your reproduction steps again maybe

**Comment by HarriBuh** [comment]
> Are we all using microG? @HarriBuh Please explain your reproduction steps again maybe

Nope, I'm still using vanilla Play Services, although I deactivated all tracking modules of it.

- 1 Enable location setting on your phone. English is not my native language, so I'm struggling with the right term. Maybe you get it anyway now?
- 2 Enable location access on your browser. It's either inside the app's settings or outside via app settings> permissions> location
- 3 Visit browserleaks.com, go to "Geolocation API". It should ask you for permission instantly.
- 3 Check back on PCAPdroid: Search for "supl"


**Internal setup:**

![Screenshot_20251107-075909.jpg](https://github.com/user-attachments/assets/61c7feed-8afd-4cbb-ad64-65e8e13788bb)

![Screenshot_20251107-075940.jpg](https://github.com/user-attachments/assets/8ec1b19d-8e82-4594-a62a-0fef4cc2a16a)

**External setup:**

![Screenshot_20251107-080011.jpg](https://github.com/user-attachments/assets/b9942361-5ed0-4ef9-9764-0b832ab20098)

![Screenshot_20251107-080103.jpg](https://github.com/user-attachments/assets/54299510-2316-447e-bf4b-3c67a81161b0)

Android's location 'enabled'
![Screenshot_20251107-081647.jpg](https://github.com/user-attachments/assets/afd4e196-0898-487f-9a1c-5bb334657e8f)

**Comment by D3SOX** [comment]
German's my native language so I got you. I did this but no supl request appeared for me

**Comment by HarriBuh** [comment]
Another app to track it (although not recommended if solely for that purpose) is RethinkDNS.
![Screenshot_20251107-194253.png](https://github.com/user-attachments/assets/4b58ddfa-667f-4fd0-b771-b9128651fc25)
No special setup required, as it will naturally track every single connection.
This time, I used Google Maps and let it track my location.

I think everybody acknowledged it by now. Next move is how to certainly change it.

**Comment by owocado** [comment]
I'm also on vanilla Play Services (meaning to move to MicroG but lazy)
I still dont see request made to supl.google.com nor to supl.grapheneos.com

However, after uninstalling this module + rebooting then following the steps mentioned by @HarriBuh above, now I see 3 requests being made to supl.google.com in PCAPDroid which is kinda weird.
